src/main/java/org/folio/rest/impl/base/AbstractEmail.java

```diff
@@ -145,6 +147,19 @@ protected Future<JsonObject> checkConfiguration(JsonObject conf, EmailEntity ent
     return promise.future();
   }
 
+  protected Future<JsonObject> checkConfiguration(JsonObject conf) {
```

Comment:
```suggestion
protected Future<JsonObject> checkConfiguration(JsonObject conf, Consumer<String> actionInCaseOfFailure) {
```
How about having only one checkConfiguration method, and passing consumer that is going to execute if something goes wrong?